packages/shared/components/ActivityRow.svelte

```diff
@@ -1,31 +1,87 @@
+
+    $: senderAccount = senderAddress
+        ? $accounts?.find((acc) => acc.addresses.some((add) => senderAddress === add.address)) ?? null
+        : null
+
+    $: receiverAccount = receiverAddresses?.length
+        ? $accounts.find((acc) => acc.addresses.some((add) => receiverAddresses.includes(add.address))) ?? null
+        : null
+
+    let initialsColor
+    let accountAlias = ''
+
+    $: {
+        const acc = payload.data.essence.data.incoming ? receiverAccount : senderAccount
+
+        if (acc) {
+            if (includeFullSender) {
+                accountAlias = acc.alias
+            }
+            if (payload.data.essence.data.internal) {
+                initialsColor = acc.color
+            }
+        }
+    }
+
+    let direction
+    $: {
+        if (includeFullSender) {
+            direction = confirmed
+                ? payload.data.essence.data.incoming
+                    ? 'general.receivedTo'
+                    : 'general.sentFrom'
+                : payload.data.essence.data.incoming
+                ? 'general.receivingTo'
+                : 'general.sendingFrom'
+        } else {
+            direction = confirmed
+                ? payload.data.essence.data.incoming
+                    ? 'general.received'
+                    : 'general.sent'
+                : payload.data.essence.data.incoming
+                ? 'general.receiving'
+                : 'general.sending'
+        }
+    }
+
     export let onClick = () => {}
 </script>
 
 <button
     on:click={onClick}
     data-label="transaction-row"
     class="w-full text-left flex rounded-2xl items-center bg-gray-100 dark:bg-gray-900 dark:bg-opacity-50 p-4 {!confirmed ? 'opacity-50' : ''}">
-    <Icon
-        boxed
-        classes="text-white dark:text-{payload.data.essence.data.internal ? 'gray-500' : `${color}-${payload.data.essence.data.incoming ? '500' : '600'}`}"
-        boxClasses="bg-{payload.data.essence.data.internal ? 'gray-500' : `${color}-${payload.data.essence.data.internal ? '500' : '600'}`} dark:bg-gray-900"
-        icon={payload.data.essence.data.internal ? 'transfer' : payload.data.essence.data.incoming ? 'chevron-down' : 'chevron-up'} />
+    <div class="w-8">
+        <Icon
+            boxed
+            classes="text-white dark:text-{payload.data.essence.data.internal ? 'gray-500' : `${color}-${payload.data.essence.data.incoming ? '500' : '600'}`}"
+            boxClasses="bg-{initialsColor ? `${initialsColor}-500` : payload.data.essence.data.internal ? 'gray-500' : `${color}-${payload.data.essence.data.internal ? '500' : '600'}`} dark:bg-gray-900"
+            icon={payload.data.essence.data.internal ? 'transfer' : payload.data.essence.data.incoming ? 'chevron-down' : 'chevron-up'} />
+    </div>
     <div class="flex flex-col ml-3.5 space-y-1.5">
-        <Text type="p" bold smaller>{truncateString(id)}</Text>
+        <Text type="p" bold smaller>{locale(direction)} {accountAlias}</Text>
```

Comment:
```suggestion
        <Text type="p" bold smaller>{locale(direction, { values: { account: accountAlias } })}</Text>
```
I'm not a linguistics expert but I believe SOV languages may order this differently, so we should leave the order up to the translator.

Reply:
Agreed :)